cogs/general.py

```diff
@@ -210,30 +224,80 @@ async def changecolor(self, context, color):
+                    if contrast > limit:
+                        chosenColor = randomColor
+                        break
+
+                color_button = Button(label="New Color", style=nextcord.ButtonStyle.blurple)    # creates post-embed button
+
+                async def color_callback(interaction):
+                    """
+                    Method to execute when the "new color" button is clicked 
+                    """
+                    if interaction.user == context.author:  # checks that user interacting with button is command sender
+                        while True:
+                            randomColor = self.generateRandomColor()
+                            contrast = self.contrast("#36393f", randomColor)
+                            if contrast > limit:
+                                break
+                            
+                        topRole = userRoles[-1]
+                        await topRole.edit(colour=nextcord.Colour(int(randomColor.replace("#", ""), 16)))   # changes top role of user
+
+                        newEmbed = nextcord.Embed ( # create new embed
+                            title="Success!",
+                            description="Color has been changed! The hex value is: " + randomColor + ". The contrast ratio is: " + str(round(contrast, 4)) + ":1",
+                            color=int(randomColor.replace("#", ""), 16)
+                        )
+                        await interaction.message.edit(embed=newEmbed)  # modify existing embed
+                
+            else:   # if message value after !changecolor is not "random"
+                contrast = self.contrast("#36393f", color)
+                chosenColor = color
+                if contrast < limit:
+                    embed = nextcord.Embed(
+                        title="Error",
+                        description="Color does not have enough contrast. That color has a contrast ratio of: " + str(round(contrast, 4)) + ":1. It needs to be above " + limit + ":1.",
+                        color=int(color.replace("#", ""), 16)
+                    )
+                    await context.send(embed=embed)
+                    return
+                    
             if len(userRoles) > 1:
                 topRole = userRoles[-1]
-                await topRole.edit(colour=nextcord.Colour(int(color.replace("#", ""), 16)))
+                await topRole.edit(colour=nextcord.Colour(int(chosenColor.replace("#", ""), 16)))
+
                 embed = nextcord.Embed(
                     title="Success!",
-                    description="Color has been changed! The contrast it has is " + str(round(contrast, 4)) + ":1",
-                    color=int(color.replace("#", ""), 16)
+                    description="Color has been changed! The hex value is: " + chosenColor + ". The contrast ratio is: " + str(round(contrast, 4)) + ":1",
+                    color=int(chosenColor.replace("#", ""), 16)

```

Comment:
The building of the embeds seems sort of done over and over, is there a way to abstract it so there is minimal repeating code?